mobile/node.go

```diff
@@ -295,7 +297,9 @@ func NewNodeWithConfig(config *NodeConfig, password string, mnemonic string) (*N
 	ncfg := ipfs.PrepareIPFSConfig(r, ignoredURI, config.Testnet, config.Testnet)
 	ncfg.Routing = constructMobileRouting
 
-	return &Node{OpenBazaarNode: core.Node, config: *config, ipfsConfig: ncfg, apiConfig: apiConfig}, nil
+	core.Node.PublishLock.Lock()
+
+	return &Node{OpenBazaarNode: core.Node, config: *config, ipfsConfig: ncfg, apiConfig: apiConfig, mtx: sync.Mutex{}}, nil
```

Comment:
Can we avoid using the global node and only use this Node instance? I'm concerned we're using a global object state which does not match this instance state? (Why do we have a global `core.Node` anyway?)

Reply:
It was a hack to get access to the lock from the main package which detects the sigint shutdown.

Reply:
We can fix it here but will need to be fixed in the main package before getting rid of it for the desktop.